File: src/main/java/com/example/springproject/Entities/RendezVous.java
package com.example.springproject.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RendezVous {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nomDeClient;
    private String prenomDeClient;
    private int tel;
    private String nomDEntreprise;
    private String adresse;
    private String ville;
    private int zip;
    private int numeroFiscale;
    private String commentaire;
    private Date dateDeRendezVous;

    @ManyToOne
    @JoinColumn(name = "employee_id", nullable = false)
   //jsonignoree
    private Employee employee;

    @Enumerated(EnumType.STRING)
    private Type type;


    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status = Status.PAS_TRAITER;
}

File: src/app/models/RendezVous.ts
import {Employee} from "./employee";


export enum Type {
  ISO_9000 ='ISO_9000',
  ISO_26030 ='ISO_26030',
  OHSAS_18001 ='OHSAS_18001',
  ISO_14001 ='ISO_14001'

}
export enum Status {
  EN_COURS = 'EN_COURS',
  PAS_TRAITER = 'PAS_TRAITER',
  ANNULE = 'ANNULE',
  MANQUE = 'MANQUE',
  CONFIRME = 'CONFIRME'
}
export interface RendezVous {
  id: number;
  nomDeClient: string;
  prenomDeClient: string;
  tel: number;
  nomDEntreprise: string;
  adresse: string;
  ville: string;
  zip: number;
  numeroFiscale: number;
  commentaire: string;
  dateDeRendezVous: Date;
  employee: Employee;
  type: Type;
  status: Status;
}
